File: ValidPalindrome/ValidPalindrome.cpp
#include <string>

using std::string;

class Solution {
public:
	bool isPalindrome(string s) {
		if (s.size() == 0)
			return true;
		for (int i = 0, j = s.size() - 1; i < j; i++, j--) {
			while (isalnum(s.at(i)) == false && i < j)
				i++;
			while (isalnum(s.at(j)) == false && i < j)
				j--;
			if (toupper(s.at(i)) != toupper(s.at(j)))
				return false;
		}
		return true;
	}
};

File: ValidPalindrome/ValidPalindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        if len(s) == 0:
            return True
        i = 0
        j = len(s) - 1
        while i < j:
            if s[i].isalnum() is False and i < j:
                i += 1
            elif s[j].isalnum() is False and i < j:
                j -= 1
            elif s[i].upper() != s[j].upper():
                return False
            else:
                i += 1
                j -= 1
        return True